spec/requests/api/supporters_request_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# License: AGPL-3.0-or-later WITH WTO-AP-3.0-or-later
+# Full license explanation at https://github.com/houdiniproject/houdini/blob/master/LICENSE
+require 'rails_helper'
+
+RSpec.describe 'Api::SupportersController', type: :request do
+	let(:supporter) { create(:supporter_with_fv_poverty) }
+	let(:nonprofit) { supporter.nonprofit }
+	let(:user) { create(:user) }
+
+	before do
+		supporter
+		user.roles.create(name: 'nonprofit_associate', host: nonprofit)
+	end
+
+	describe 'GET /' do
+		let(:nonprofit) { Nonprofit.first }
+
+		it 'returns http success when logged in' do
+			sign_in user
+			get "/api/nonprofits/#{nonprofit.id}/supporters"
+			expect(response).to have_http_status(:success)
+		end
+
+		it 'returns http success when not logged in' do
+			get "/api/nonprofits/#{nonprofit.id}/supporters"
+			expect(response).to have_http_status(:unauthorized)
+		end
+	end
+
+	describe 'GET /:id' do
+		it 'returns http success when logged in' do
+			sign_in user
+			get "/api/nonprofits/#{nonprofit.id}/supporters/#{supporter.id}"
+			expect(response).to have_http_status(:success)
+		end
+
+		it 'returns http success when not logged in' do
+			get "/api/nonprofits/#{nonprofit.id}/supporters/#{supporter.id}"
+			expect(response).to have_http_status(:unauthorized)
+		end
+	end
```

Comment:
Wouldn't it be a good idea to test how the return looks like? I think it also helps as a documentation for the API.

Reply:
Probably.

There's multiple schools of thought in the Rails community on that but it wouldn't hurt to do that. :+1:

Reply:
I've added this to the specs!